machine_learning/benchmarks/vqc_base_benchmark.py

```diff
@@ -0,0 +1,147 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base for VQC based classifier benchmarks."""
+from abc import ABC
+from typing import Optional, Tuple
+
+import numpy as np
+from qiskit.algorithms.optimizers import Optimizer
+from qiskit.circuit.library import RealAmplitudes, ZZFeatureMap
+from qiskit.utils import algorithm_globals
+from qiskit_machine_learning.algorithms import NeuralNetworkClassifier, VQC
+from sklearn.datasets import load_iris
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler, OneHotEncoder
+
+from .base_classifier_benchmark import BaseClassifierBenchmark
+from .datasets import (
+    DATASET_SYNTHETIC_CLASSIFICATION_FEATURES,
+    DATASET_SYNTHETIC_CLASSIFICATION_LABELS,
+)
+
+
+class VqcBaseClassifierBenchmark(BaseClassifierBenchmark, ABC):
+    """Base for Opflow Classifier benchmarks."""
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # prepare synthetic
+        (
+            synth_train_features,
+            synth_test_features,
+            synth_train_labels,
+            synth_test_labels,
+        ) = train_test_split(
+            DATASET_SYNTHETIC_CLASSIFICATION_FEATURES,
+            DATASET_SYNTHETIC_CLASSIFICATION_LABELS,
+            test_size=5,
+            shuffle=False,
+        )
+
+        # one hot encoding for VQC
+        encoder = OneHotEncoder()
+        # VQC does not work with csr_matrix returned by the encoder
+        synth_train_labels = encoder.fit_transform(synth_train_labels.reshape(-1, 1)).toarray()
+        synth_test_labels = encoder.fit_transform(synth_test_labels.reshape(-1, 1)).toarray()
+
+        # prepare iris
+        iris_features, iris_labels = self._prepare_iris()
+
+        (
+            iris_train_features,
+            iris_test_features,
+            iris_train_labels,
+            iris_test_labels,
+        ) = train_test_split(
+            iris_features,
+            iris_labels,
+            test_size=5,
+            shuffle=False,
+        )
+
+        self.datasets = {
+            "dataset_synthetic": {
+                "train_features": synth_train_features,
+                "train_labels": synth_train_labels,
+                "test_features": synth_test_features,
+                "test_labels": synth_test_labels,
+            },
+            "dataset_iris": {
+                "train_features": iris_train_features,
+                "train_labels": iris_train_labels,
+                "test_features": iris_test_features,
+                "test_labels": iris_test_labels,
+            },
+        }
+
+    def _prepare_iris(self) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
I have seen similar methods in other classes. Can we avoid some code duplication by adding more arguments and reusing the general method in other classes?

Reply:
I know, they are similar. I though about this. The methods are slightly different though:
- iris in CircuitQNN is sort of basic, 3 classes, labels {0, 1}
- iris in Opflow subsets only 2 classes, labels {-1, 1}
- iris in VQC, 3 classes and one hot encoding.

So they can be generalized, let me try it out and check if such a version looks good.

Reply:
> I have seen similar methods in other classes. Can we avoid some code duplication by adding more arguments and reusing the general method in other classes?

OK, re-worked how the datasets are instantiated.